lighthouse-core/gather/gather-runner.js

```diff
@@ -248,31 +242,25 @@ class GatherRunner {
     const status = 'Loading page & waiting for onload';
     log.log('status', status, gatherernames);
 
-    const pass = Promise.resolve()
-      // Clear disk & memory cache if it's a perf run
-      .then(_ => {
-        if (isPerfRun) driver.cleanBrowserCaches();
```

Comment:
this code wasn't be awaited, but it looks like it should be. Same with `beginTrace` down below.

Reply:
maybe https://palantir.github.io/tslint/rules/no-floating-promises/, which isn't perfect but would have caught this

Reply:
I think just using async/await makes these stand out a lot more. That lint rule might be more trouble than its worth.